### ISSUE TITLE
Unauthorized Access on fastlane init

### ISSUE THREAD

**Issue by stevekohls** [fastlane/fastlane#6084]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I have reviewed the existing github issues relating to this error, and none of them seem to show that this issue has been recently noted, or solved. Others address incorrect retry handling, or lack of two-factor authentication support, or other unrelated issues.

I am initializing  a new project and I am getting an "Unauthorized Access" error. I have verified I am using the latest versions of fastlane and all dependencies, as well as the Xcode command-line tools. fastlane does not crash and I am able to continue with the manual steps once access fails.

I am able to log in to the Developer portal using the same credentials I enter via fastlane. I have two-factor authentication enabled, and I do receive the 6 digit verification code and enter that correctly when prompted.

I am not using an Enterprise account, and the Apple ID I am using is only tied to one developer team. Also, I have the Admin role in my team, and the `team_id` in the generated __Appfile__ is correct.

I made sure to go back and clear all of the following information before capturing the output below.

 - delete the `deliver` entry from the keychain
 - delete the `spaceship*` files from `/tmp`
 - delete `~/.spaceship`
 - delete the `fastlane` directory from the project directory

##### Complete output when running fastlane, including the stack trace and command used

```
$ fastlane init --verbose
DEBUG [2016-09-12 15:50:06.42]: Checking if there are any plugins that should be loaded...
INFO [2016-09-12 15:50:06.43]: Detected iOS/Mac project in current directory...
INFO [2016-09-12 15:50:06.43]: This setup will help you get up and running in no time.
INFO [2016-09-12 15:50:06.43]: fastlane will check what tools you're already using and set up
INFO [2016-09-12 15:50:06.43]: the tool automatically for you. Have fun! 
INFO [2016-09-12 15:50:06.43]: Created new folder './fastlane'.
INFO [2016-09-12 15:50:06.43]: $ xcodebuild clean -showBuildSettings -project ./FastlaneTest.xcodeproj
Your Apple ID (e.g. fastlane@krausefx.com): xxx@xxx.com
WARN [2016-09-12 15:50:14.64]: Verifying if app is available on the Apple Developer Portal and iTunes Connect...
INFO [2016-09-12 15:50:14.64]: Starting login with user 'xxx@xxx.com'
-------------------------------------------------------------------------------------
The login information you enter will be stored in your Mac OS Keychain
You can also pass the password using the `FASTLANE_PASSWORD` env variable
More information about it on GitHub: https://github.com/fastlane/fastlane/tree/maxxxr/credentials_manager
-------------------------------------------------------------------------------------
Password (for xxx@xxx.com): ********
Two Factor Authentication for account 'xxx@xxx.com' is enabled
Please enter the 6 digit code: xxxxxx
Requesting session...
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
Auth error received: 'Unauthorized Access'. Login in again then retrying after 3 seconds (remaining: 4)...
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
Auth error received: 'Unauthorized Access'. Login in again then retrying after 3 seconds (remaining: 3)...
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
Auth error received: 'Unauthorized Access'. Login in again then retrying after 3 seconds (remaining: 2)...
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
Auth error received: 'Unauthorized Access'. Login in again then retrying after 3 seconds (remaining: 1)...
Loading session from '/Users/xxx/.spaceship/xxx@xxx.com/cookie'
ERROR [2016-09-12 15:50:57.78]: Unauthorized Access
ERROR [2016-09-12 15:50:57.78]: An error occured during the setup process. Falling back to manual setup!
App Identifier (com.krausefx.app): com.xxx.FastlaneTest
INFO [2016-09-12 15:51:09.77]: Created new file './fastlane/Appfile'. Edit it to manage your preferred app metadata information.
Would you like to create your app on iTunes Connect and the Developer Portal? (y/n)
n
Optional: The scheme name of your app (If you don't need one, just hit Enter): 
WARN [2016-09-12 15:51:14.35]: 'snapshot' not enabled.
WARN [2016-09-12 15:51:14.35]: 'cocoapods' not enabled.
WARN [2016-09-12 15:51:14.35]: 'carthage' not enabled.
INFO [2016-09-12 15:51:14.35]: Created new file './fastlane/Fastfile'. Edit it to manage your own deployment lanes.
INFO [2016-09-12 15:51:14.35]: fastlane will send the number of errors for each action to
INFO [2016-09-12 15:51:14.35]: https://github.com/fastlane/enhancer to detect integration issues
INFO [2016-09-12 15:51:14.35]: No sensitive/private information will be uploaded
```

If I delete the `fastlane` directory in my project and run `fastlane init --verbose` again, I get the same results, I'm just not asked for my two-factor authentication info.

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

The `Fastfile` is the default generated by `fastlane init`

Fastfile:
```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.102.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    
    
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym # Build your app - more options available
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```

### Environment

fastlane version (run `fastlane -v`): __1.102.0__

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? __NO__

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? __NO__

**Comment by stevekohls** [comment]
Regarding the comment above by __fastlane-bot__, this is not related to code signing.

**Comment by KrauseFx** [comment]
That is weird, you can run `spaceship -u [your email]` to quickly try the login without triggering all of fastlane. 

Can you confirm that you can login on the following services:

- Apple Developer Portal, and list all certificates
- iTunes Connect
- Login in Xcode in the settings

**Comment by stevekohls** [comment]
Hi @KrauseFx 

When I execute `spaceship -u [my email]`, it does successfully log me in to iTunes Connect and the Developer Portal.

I can log in to the Developer Portal, and login via Xcode works as well.

When I go to log in to iTunes Connect, it shows me a "Getting Started" page, instead of the one I'm used to seeing, when used with other accounts and teams. I believe the problem may be that I am not the Team Agent, and I may not have been granted any Roles yet.

I'm following up on the Roles issue now.

**Comment by stevekohls** [comment]
I was able to verify with one of my co-workers, and he can see me on iTunes Connect, and I am listed as an Admin. Digging further now....

**Comment by stevekohls** [comment]
I contacted Apple Developer Support to see if they can fix things. I've seen some mentions on Stack Overflow and elsewhere that other people had the same problems. Contacting DTS fixed the problem.

I don't know if this is a red herring or not. We'll see after I hear back from Apple.

**Comment by stevekohls** [comment]
@KrauseFx here's an update:

I had my co-worker double-check, and I am __NOT__ a user on iTunes Connect. Even so, if I execute `spaceship -u [my email]` I see the following output:

```
Logging into to iTunes Connect (xxx@xxx.com)...
Successfully logged in to iTunes Connect

Logging into the Developer Portal (xxx@xxx.com)...
Successfully logged in to the Developer Portal

---------------------------------------
| Welcome to the spaceship playground |
---------------------------------------

Enter docs to open up the documentation
Enter exit to exit the spaceship playground
Enter _ to access the return value of the last executed command

Just enter the commands and confirm with Enter
[1] pry(main)> 
```

Within `spaceship` if I execute `Spaceship::Tunes::Application.all` I see an `Unauthorized Access` error, as I would expect.

```
[6] pry(main)> Spaceship::Tunes::Application.all
Spaceship::Client::UnauthorizedAccessError: Unauthorized Access
from /usr/local/lib/ruby/gems/2.3.0/gems/spaceship-0.32.1/lib/spaceship/client.rb:428:in `block in send_request'
```

It seems that there are two problems here:

1. `spaceship -u [my email]` incorrectly reports `Successfully logged in to the Developer Portal` for a user who is not in iTunes Connect.
2. `fastlane init --verbose` correctly displays an `Unauthorized Access` error, however for better debugging, it should report _which_ service is reporting the access error.

I have requested to my Team Agent to be added to iTunes Connect. Once that is done, I'll test this all again.

**Comment by stevekohls** [comment]
So I'm in iTunes Connect now, and everything is working great. I'm closing this issue and will be opening two more for the items I mentioned above.

**Comment by TKBurner** [comment]
@stevekohls Thanks Steve! Really appreciate the update. You are required to be a iTC admin for spaceship to work as expected. Thanks!
